phase1-coordinator/src/objects/round.rs

```diff
@@ -726,24 +629,109 @@ impl Round {
                 )?;
             }
             Participant::Verifier(_) => {
-                // Initialize the next challenge file.
-                storage.initialize(
-                    Locator::ContributionFile(locked_locators.next_contribution.clone()),
-                    Object::contribution_file_size(environment, chunk_id, true),
-                )?;
-
-                // Initialize the contribution file signature.
-                storage.initialize(
-                    Locator::ContributionFileSignature(locked_locators.next_contribution_file_signature.clone()),
-                    Object::contribution_file_signature_size(true),
-                )?;
+                return Err(CoordinatorError::ExpectedContributor);
             }
         };
 
         debug!("{} locked chunk {}", participant, chunk_id);
         Ok(locked_locators)
     }
 
+    /// Initialize the files for the next challenge
+    pub fn initialize_verifier_response_files(
+        &self,
+        environment: &Environment,
+        storage: &mut impl Storage,
+        participant: &Participant,
+        chunk_id: u64,
+        locators: &LockedLocators,
+    ) -> Result<(), CoordinatorError> {
+        if !participant.is_verifier() {
+            return Err(CoordinatorError::ExpectedVerifier);
+        }
+        // Initialize the next challenge file.
+        storage.initialize(
```

Comment:
I think this could benefit from `InitializeAction` https://github.com/AleoHQ/aleo-setup/blob/01ee29f3ec74ee8a1068633f649f0fbb88149101/phase1-coordinator/src/objects/round.rs#L721 in #367, to avoid mutating `storage` in this module, we could keep an eye on which one gets merged first and implement it

Reply:
This is a bit controversial. In #367 do you want to replace a direct call which does the change with an inderect call, which performs the changes somewhat later? I want to see an explicit call here.

Reply:
It makes this closer to a pure function, which makes it more predictable and easier to test